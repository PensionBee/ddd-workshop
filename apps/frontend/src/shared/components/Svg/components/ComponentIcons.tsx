export { ReactComponent as Check } from "@assets/icons/component-icons/checkmark.svg";
