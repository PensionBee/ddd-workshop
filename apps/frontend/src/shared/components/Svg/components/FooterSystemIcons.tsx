/* eslint-disable react-refresh/only-export-components */
export { default as FscsLogo } from "@assets/images/fscs-protected-colour.svg";
export { default as TrustPilot } from "@assets/images/trustpilot.svg";
export { default as InternetCrystalMark } from "@assets/images/internet-crystal-mark.svg";
