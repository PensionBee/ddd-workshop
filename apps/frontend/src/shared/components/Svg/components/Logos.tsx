export { default as PensionBeeLogo } from "@assets/logos/pensionbee-logo.svg";
export { default as PensionBeeLogoNegative } from "@assets/logos/pensionbee-logo-negative.svg";
export { default as PensionBeeLogoBorder } from "@assets/logos/pensionbee-logo-border.svg";
