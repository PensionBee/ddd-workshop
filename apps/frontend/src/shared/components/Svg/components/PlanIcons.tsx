export { ReactComponent as Tailored } from "@assets/icons/plan-icons/tailored.svg";
export { ReactComponent as FourPlus } from "@assets/icons/plan-icons/4plus.svg";
export { ReactComponent as Preserve } from "@assets/icons/plan-icons/preserve.svg";
export { ReactComponent as Shariah } from "@assets/icons/plan-icons/shariah.svg";
export { ReactComponent as Tracker } from "@assets/icons/plan-icons/tracker.svg";
export { ReactComponent as Impact } from "@assets/icons/plan-icons/impact.svg";
export { ReactComponent as FossilFuelFree } from "@assets/icons/plan-icons/fossil-fuel-free.svg";
export { ReactComponent as PreAnnuity } from "@assets/icons/plan-icons/pre-annuity.svg";
