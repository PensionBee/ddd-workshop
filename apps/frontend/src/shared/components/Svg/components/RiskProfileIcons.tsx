export { ReactComponent as Higher } from "@assets/icons/risk-profile-icons/higher.svg";
export { ReactComponent as Medium } from "@assets/icons/risk-profile-icons/medium.svg";
export { ReactComponent as Lower } from "@assets/icons/risk-profile-icons/lower.svg";
