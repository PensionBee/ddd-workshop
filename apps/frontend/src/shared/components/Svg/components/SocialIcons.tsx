export { ReactComponent as Facebook } from "@assets/icons/social-icons/facebook-active.svg";
export { ReactComponent as Instagram } from "@assets/icons/social-icons/instagram-active.svg";
export { ReactComponent as LinkedIn } from "@assets/icons/social-icons/linkedin-active.svg";
export { ReactComponent as Podcast } from "@assets/icons/social-icons/podcast-active.svg";
export { ReactComponent as Twitter } from "@assets/icons/social-icons/twitter-active.svg";
export { ReactComponent as Youtube } from "@assets/icons/social-icons/youtube-active.svg";
