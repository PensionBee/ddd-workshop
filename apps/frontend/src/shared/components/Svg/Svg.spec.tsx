import { render } from "@shared/testUtils";
import React from "react";
import * as ComponentIcons from "./components/ComponentIcons";
import * as FooterSystemIcons from "./components/FooterSystemIcons";
import * as LargeSystemIcons from "./components/LargeSystemIcons";
import * as Logos from "./components/Logos";
import * as SmallSystemIcons from "./components/SmallSystemIcons";
import * as SocialIcons from "./components/SocialIcons";
import * as PlanIcons from "./components/PlanIcons";
import * as RiskProfileIcons from "./components/RiskProfileIcons";

import type { SvgProps } from "./Svg";
import {
  ComponentIcon,
  FooterSystemIcon,
  LargeSystemIcon,
  LogoSvg,
  SmallSystemIcon,
  SocialIcon,
  PlanIcon,
  RiskProfileIcon,
} from "./Svg";

/* Helper functions */
const fromContainer = (container: HTMLElement) => ({
  getSVGs: () => container.getElementsByTagName("svg"),
});

const shouldRenderSvg = (Icons: object) => {
  it("should render svg correctly", () => {
    const IconComponents = Object.entries(Icons);
    const { container, asFragment } = render(
      <>
        {IconComponents.map(([iconName, Svg]) => (
          <Svg key={iconName} data-rc={iconName} />
        ))}
      </>
    );
    const { getSVGs } = fromContainer(container);

    expect(getSVGs().length).toBe(IconComponents.length);
    expect(asFragment()).toMatchSnapshot();
  });
};

const shouldRenderComponent = <T extends string>(
  Component: React.FC<SvgProps<T>>,
  icon: SvgProps<T>["icon"]
) => {
  it("should render component correctly", () => {
    const { container, asFragment } = render(<Component icon={icon} />);
    const { getSVGs } = fromContainer(container);

    expect(getSVGs().length).toBe(1);
    expect(asFragment()).toMatchSnapshot();
  });
};

describe("Svg", () => {
  describe("ComponentIcons", () => shouldRenderSvg(ComponentIcons));
  describe("FooterSystemIcons", () => shouldRenderSvg(FooterSystemIcons));
  describe("LargeSystemIcons", () => shouldRenderSvg(LargeSystemIcons));
  describe("Logos", () => shouldRenderSvg(Logos));
  describe("SmallSystemIcons", () => shouldRenderSvg(SmallSystemIcons));
  describe("SocialIcons", () => shouldRenderSvg(SocialIcons));
  describe("PlanIcons", () => shouldRenderSvg(PlanIcons));
  describe("RiskProfileIcons", () => shouldRenderSvg(RiskProfileIcons));

  describe("ComponentIcon", () =>
    shouldRenderComponent(ComponentIcon, "Check"));
  describe("FooterSystemIcon", () =>
    shouldRenderComponent(FooterSystemIcon, "FscsLogo"));
  describe("LargeSystemIcon", () =>
    shouldRenderComponent(LargeSystemIcon, "AndroidId"));
  describe("LogoSvg", () => shouldRenderComponent(LogoSvg, "PensionBeeLogo"));
  describe("SmallSystemIcon", () =>
    shouldRenderComponent(SmallSystemIcon, "Actions"));
  describe("SocialIcon", () => shouldRenderComponent(SocialIcon, "Facebook"));
  describe("PlanIcon", () => shouldRenderComponent(PlanIcon, "Tailored"));
  describe("RiskProfileIcon", () =>
    shouldRenderComponent(RiskProfileIcon, "Medium"));
});
