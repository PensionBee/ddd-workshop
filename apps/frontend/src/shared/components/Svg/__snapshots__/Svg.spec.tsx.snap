// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Svg ComponentIcon should render component correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe lmcdLK"
    data-rc="ComponentIcon.Check"
  >
    <svg>
      checkmark.svg
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Svg ComponentIcons should render svg correctly 1`] = `
<DocumentFragment>
  <svg
    data-rc="Check"
  >
    checkmark.svg
  </svg>
</DocumentFragment>
`;

exports[`Svg FooterSystemIcon should render component correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe lmcdLK"
    data-rc="FooterSystemIcon.FscsLogo"
  >
    <svg>
      fscs-protected-colour.svg
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Svg FooterSystemIcons should render svg correctly 1`] = `
<DocumentFragment>
  <svg
    data-rc="FscsLogo"
  >
    fscs-protected-colour.svg
  </svg>
  <svg
    data-rc="InternetCrystalMark"
  >
    internet-crystal-mark.svg
  </svg>
  <svg
    data-rc="TrustPilot"
  >
    trustpilot.svg
  </svg>
</DocumentFragment>
`;

exports[`Svg LargeSystemIcon should render component correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe lmcdLK"
    data-rc="LargeSystemIcon.AndroidId"
  >
    <svg>
      android-id.svg
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Svg LargeSystemIcons should render svg correctly 1`] = `
<DocumentFragment>
  <svg
    data-rc="Account"
  >
    account.svg
  </svg>
  <svg
    data-rc="ActionRequired"
  >
    action-required.svg
  </svg>
  <svg
    data-rc="AddPension"
  >
    add-pension.svg
  </svg>
  <svg
    data-rc="Address"
  >
    address.svg
  </svg>
  <svg
    data-rc="Address2"
  >
    address-2.svg
  </svg>
  <svg
    data-rc="AlertInfo"
  >
    alert-info.svg
  </svg>
  <svg
    data-rc="AlertWarning"
  >
    alert-warning.svg
  </svg>
  <svg
    data-rc="Analytics"
  >
    analytics.svg
  </svg>
  <svg
    data-rc="AndroidId"
  >
    android-id.svg
  </svg>
  <svg
    data-rc="ArrowDown"
  >
    arrow-down.svg
  </svg>
  <svg
    data-rc="ArrowLeft"
  >
    arrow-left.svg
  </svg>
  <svg
    data-rc="ArrowRight"
  >
    arrow-right.svg
  </svg>
  <svg
    data-rc="ArrowUp"
  >
    arrow-up.svg
  </svg>
  <svg
    data-rc="Balance"
  >
    balance.svg
  </svg>
  <svg
    data-rc="BalancePot"
  >
    balance-pot.svg
  </svg>
  <svg
    data-rc="Bank"
  >
    bank.svg
  </svg>
  <svg
    data-rc="Bar"
  >
    bar.svg
  </svg>
  <svg
    data-rc="Bee"
  >
    bee.svg
  </svg>
  <svg
    data-rc="Bubble"
  >
    bubble.svg
  </svg>
  <svg
    data-rc="Building"
  >
    building.svg
  </svg>
  <svg
    data-rc="Calculator"
  >
    calculator.svg
  </svg>
  <svg
    data-rc="Calendar"
  >
    calendar.svg
  </svg>
  <svg
    data-rc="Card"
  >
    card.svg
  </svg>
  <svg
    data-rc="Categories"
  >
    categories.svg
  </svg>
  <svg
    data-rc="Chat"
  >
    chat.svg
  </svg>
  <svg
    data-rc="ChevronDown"
  >
    chevron-down.svg
  </svg>
  <svg
    data-rc="ChevronYellow"
  >
    chevron-yellow.svg
  </svg>
  <svg
    data-rc="Clock"
  >
    clock.svg
  </svg>
  <svg
    data-rc="Close"
  >
    close.svg
  </svg>
  <svg
    data-rc="ContactDetail"
  >
    contact-detail.svg
  </svg>
  <svg
    data-rc="Contribute"
  >
    contribute.svg
  </svg>
  <svg
    data-rc="Customer"
  >
    customer.svg
  </svg>
  <svg
    data-rc="Document"
  >
    document.svg
  </svg>
  <svg
    data-rc="Download"
  >
    download.svg
  </svg>
  <svg
    data-rc="Dropdown"
  >
    dropdown.svg
  </svg>
  <svg
    data-rc="Edit"
  >
    edit.svg
  </svg>
  <svg
    data-rc="Email"
  >
    email.svg
  </svg>
  <svg
    data-rc="Email2"
  >
    email2.svg
  </svg>
  <svg
    data-rc="Employed"
  >
    employed.svg
  </svg>
  <svg
    data-rc="Employer"
  >
    employer.svg
  </svg>
  <svg
    data-rc="Equal"
  >
    equal.svg
  </svg>
  <svg
    data-rc="Expand"
  >
    expand.svg
  </svg>
  <svg
    data-rc="Eye"
  >
    eye.svg
  </svg>
  <svg
    data-rc="Eyeclosed"
  >
    eyeclosed.svg
  </svg>
  <svg
    data-rc="Faceid"
  >
    faceid.svg
  </svg>
  <svg
    data-rc="Filter"
  >
    filter.svg
  </svg>
  <svg
    data-rc="Fingerprint"
  >
    fingerprint.svg
  </svg>
  <svg
    data-rc="Flexible"
  >
    flexible.svg
  </svg>
  <svg
    data-rc="Freeze"
  >
    freeze.svg
  </svg>
  <svg
    data-rc="HMRC"
  >
    hmrc.svg
  </svg>
  <svg
    data-rc="Heart"
  >
    heart.svg
  </svg>
  <svg
    data-rc="Heart2"
  >
    heart-2.svg
  </svg>
  <svg
    data-rc="Idea"
  >
    idea.svg
  </svg>
  <svg
    data-rc="Info"
  >
    info.svg
  </svg>
  <svg
    data-rc="Link"
  >
    link.svg
  </svg>
  <svg
    data-rc="Lock"
  >
    lock.svg
  </svg>
  <svg
    data-rc="Lock2"
  >
    lock-2.svg
  </svg>
  <svg
    data-rc="LogOut"
  >
    log-out.svg
  </svg>
  <svg
    data-rc="LongArrowDown"
  >
    long-arrow-down.svg
  </svg>
  <svg
    data-rc="LongArrowUp"
  >
    long-arrow-up.svg
  </svg>
  <svg
    data-rc="LtdCompany"
  >
    ltd-company.svg
  </svg>
  <svg
    data-rc="Menu"
  >
    menu.svg
  </svg>
  <svg
    data-rc="Minus"
  >
    minus.svg
  </svg>
  <svg
    data-rc="MinusYellow"
  >
    minus-yellow.svg
  </svg>
  <svg
    data-rc="Passcode"
  >
    passcode.svg
  </svg>
  <svg
    data-rc="Password"
  >
    password.svg
  </svg>
  <svg
    data-rc="Pathways"
  >
    pathways.svg
  </svg>
  <svg
    data-rc="Pause"
  >
    pause.svg
  </svg>
  <svg
    data-rc="Pdf"
  >
    pdf.svg
  </svg>
  <svg
    data-rc="People"
  >
    people.svg
  </svg>
  <svg
    data-rc="Percentage"
  >
    percentage.svg
  </svg>
  <svg
    data-rc="Performance"
  >
    performance.svg
  </svg>
  <svg
    data-rc="Personal"
  >
    personal.svg
  </svg>
  <svg
    data-rc="Pie"
  >
    pie.svg
  </svg>
  <svg
    data-rc="Play"
  >
    play.svg
  </svg>
  <svg
    data-rc="PlusYellow"
  >
    plus-yellow.svg
  </svg>
  <svg
    data-rc="Pound"
  >
    pound.svg
  </svg>
  <svg
    data-rc="Present"
  >
    present.svg
  </svg>
  <svg
    data-rc="Projected"
  >
    projected.svg
  </svg>
  <svg
    data-rc="Quote"
  >
    quote.svg
  </svg>
  <svg
    data-rc="Resources"
  >
    resources.svg
  </svg>
  <svg
    data-rc="Search"
  >
    search.svg
  </svg>
  <svg
    data-rc="ShareAndroid"
  >
    share-android.svg
  </svg>
  <svg
    data-rc="ShareApple"
  >
    share-apple.svg
  </svg>
  <svg
    data-rc="SignatureMarker"
  >
    signature-marker.svg
  </svg>
  <svg
    data-rc="SquareStar"
  >
    square star.svg
  </svg>
  <svg
    data-rc="Star"
  >
    star.svg
  </svg>
  <svg
    data-rc="Star2"
  >
    star2.svg
  </svg>
  <svg
    data-rc="Support"
  >
    support.svg
  </svg>
  <svg
    data-rc="SwitchPlans"
  >
    switch-plans.svg
  </svg>
  <svg
    data-rc="Telephone"
  >
    telephone.svg
  </svg>
  <svg
    data-rc="Tick"
  >
    tick.svg
  </svg>
  <svg
    data-rc="TransferIn"
  >
    transfer-in.svg
  </svg>
  <svg
    data-rc="Transferring"
  >
    transferring.svg
  </svg>
  <svg
    data-rc="Trash"
  >
    trash.svg
  </svg>
  <svg
    data-rc="Types"
  >
    types.svg
  </svg>
  <svg
    data-rc="UploadDoc"
  >
    upload-doc.svg
  </svg>
  <svg
    data-rc="Video"
  >
    video.svg
  </svg>
  <svg
    data-rc="Withdraw"
  >
    withdraw.svg
  </svg>
</DocumentFragment>
`;

exports[`Svg LogoSvg should render component correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe lmcdLK"
    data-rc="LogoSvg.PensionBeeLogo"
  >
    <svg>
      pensionbee-logo.svg
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Svg Logos should render svg correctly 1`] = `
<DocumentFragment>
  <svg
    data-rc="PensionBeeLogo"
  >
    pensionbee-logo.svg
  </svg>
  <svg
    data-rc="PensionBeeLogoBorder"
  >
    pensionbee-logo-border.svg
  </svg>
  <svg
    data-rc="PensionBeeLogoNegative"
  >
    pensionbee-logo-negative.svg
  </svg>
</DocumentFragment>
`;

exports[`Svg PlanIcon should render component correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe lmcdLK"
    data-rc="PlanIcon.Tailored"
  >
    <svg>
      tailored.svg
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Svg PlanIcons should render svg correctly 1`] = `
<DocumentFragment>
  <svg
    data-rc="FossilFuelFree"
  >
    fossil-fuel-free.svg
  </svg>
  <svg
    data-rc="FourPlus"
  >
    4plus.svg
  </svg>
  <svg
    data-rc="Impact"
  >
    impact.svg
  </svg>
  <svg
    data-rc="PreAnnuity"
  >
    pre-annuity.svg
  </svg>
  <svg
    data-rc="Preserve"
  >
    preserve.svg
  </svg>
  <svg
    data-rc="Shariah"
  >
    shariah.svg
  </svg>
  <svg
    data-rc="Tailored"
  >
    tailored.svg
  </svg>
  <svg
    data-rc="Tracker"
  >
    tracker.svg
  </svg>
</DocumentFragment>
`;

exports[`Svg RiskProfileIcon should render component correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe lmcdLK"
    data-rc="RiskProfileIcon.Medium"
  >
    <svg>
      medium.svg
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Svg RiskProfileIcons should render svg correctly 1`] = `
<DocumentFragment>
  <svg
    data-rc="Higher"
  >
    higher.svg
  </svg>
  <svg
    data-rc="Lower"
  >
    lower.svg
  </svg>
  <svg
    data-rc="Medium"
  >
    medium.svg
  </svg>
</DocumentFragment>
`;

exports[`Svg SmallSystemIcon should render component correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe lmcdLK"
    data-rc="SmallSystemIcon.Actions"
  >
    <svg>
      actions.svg
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Svg SmallSystemIcons should render svg correctly 1`] = `
<DocumentFragment>
  <svg
    data-rc="Actions"
  >
    actions.svg
  </svg>
  <svg
    data-rc="Alarm"
  >
    alarm.svg
  </svg>
  <svg
    data-rc="Alert"
  >
    alert.svg
  </svg>
  <svg
    data-rc="AlertInfo"
  >
    alert-info.svg
  </svg>
  <svg
    data-rc="Analytics"
  >
    analytics.svg
  </svg>
  <svg
    data-rc="AtSign"
  >
    at-sign.svg
  </svg>
  <svg
    data-rc="Attach"
  >
    attach.svg
  </svg>
  <svg
    data-rc="Back"
  >
    back.svg
  </svg>
  <svg
    data-rc="BackArrow"
  >
    back-arrow.svg
  </svg>
  <svg
    data-rc="Backpack"
  >
    backpack.svg
  </svg>
  <svg
    data-rc="Backspace"
  >
    backspace.svg
  </svg>
  <svg
    data-rc="Backward"
  >
    backward.svg
  </svg>
  <svg
    data-rc="Bag"
  >
    bag.svg
  </svg>
  <svg
    data-rc="Balloon"
  >
    balloon.svg
  </svg>
  <svg
    data-rc="Bank"
  >
    bank.svg
  </svg>
  <svg
    data-rc="Book"
  >
    book.svg
  </svg>
  <svg
    data-rc="Bookmark"
  >
    bookmark.svg
  </svg>
  <svg
    data-rc="Bookmark2"
  >
    bookmark-2.svg
  </svg>
  <svg
    data-rc="Briefcase"
  >
    briefcase.svg
  </svg>
  <svg
    data-rc="Brush"
  >
    brush.svg
  </svg>
  <svg
    data-rc="Calculator"
  >
    calculator.svg
  </svg>
  <svg
    data-rc="CalculatorAlt"
  >
    calculator-alt.svg
  </svg>
  <svg
    data-rc="Calendar"
  >
    calendar.svg
  </svg>
  <svg
    data-rc="Call"
  >
    call.svg
  </svg>
  <svg
    data-rc="Camera"
  >
    camera.svg
  </svg>
  <svg
    data-rc="Car"
  >
    car.svg
  </svg>
  <svg
    data-rc="Card"
  >
    card.svg
  </svg>
  <svg
    data-rc="Cart"
  >
    cart.svg
  </svg>
  <svg
    data-rc="Cast"
  >
    cast.svg
  </svg>
  <svg
    data-rc="ChartBar"
  >
    chart-bar.svg
  </svg>
  <svg
    data-rc="ChartPie"
  >
    chart-pie.svg
  </svg>
  <svg
    data-rc="Chat"
  >
    chat.svg
  </svg>
  <svg
    data-rc="Chats"
  >
    chats.svg
  </svg>
  <svg
    data-rc="Check"
  >
    check.svg
  </svg>
  <svg
    data-rc="CheckboxOn"
  >
    checkbox-on.svg
  </svg>
  <svg
    data-rc="Chip"
  >
    chip.svg
  </svg>
  <svg
    data-rc="CircleCross"
  >
    circle-cross.svg
  </svg>
  <svg
    data-rc="CircleTick"
  >
    circle-tick.svg
  </svg>
  <svg
    data-rc="City"
  >
    city.svg
  </svg>
  <svg
    data-rc="Cloud"
  >
    cloud.svg
  </svg>
  <svg
    data-rc="CloudDownload"
  >
    cloud-download.svg
  </svg>
  <svg
    data-rc="CloudUpload"
  >
    cloud-upload.svg
  </svg>
  <svg
    data-rc="CoinPound"
  >
    coin-pound.svg
  </svg>
  <svg
    data-rc="Comment"
  >
    comment.svg
  </svg>
  <svg
    data-rc="Configure"
  >
    configure.svg
  </svg>
  <svg
    data-rc="Controller"
  >
    controller.svg
  </svg>
  <svg
    data-rc="Copy"
  >
    copy.svg
  </svg>
  <svg
    data-rc="Cut"
  >
    cut.svg
  </svg>
  <svg
    data-rc="Cutlery"
  >
    cutlery.svg
  </svg>
  <svg
    data-rc="Delete"
  >
    delete.svg
  </svg>
  <svg
    data-rc="Dial"
  >
    dial.svg
  </svg>
  <svg
    data-rc="Disabled"
  >
    disabled.svg
  </svg>
  <svg
    data-rc="Download"
  >
    download.svg
  </svg>
  <svg
    data-rc="Edit"
  >
    edit.svg
  </svg>
  <svg
    data-rc="EggTimer"
  >
    egg-timer.svg
  </svg>
  <svg
    data-rc="EmojiAngry"
  >
    emoji-angry.svg
  </svg>
  <svg
    data-rc="EmojiHappy"
  >
    emoji-happy.svg
  </svg>
  <svg
    data-rc="EmojiShocked"
  >
    emoji-shocked.svg
  </svg>
  <svg
    data-rc="EmojiSmile"
  >
    emoji-smile.svg
  </svg>
  <svg
    data-rc="Export"
  >
    export.svg
  </svg>
  <svg
    data-rc="Factory"
  >
    factory.svg
  </svg>
  <svg
    data-rc="File"
  >
    file.svg
  </svg>
  <svg
    data-rc="Filter"
  >
    filter.svg
  </svg>
  <svg
    data-rc="Flame"
  >
    flame.svg
  </svg>
  <svg
    data-rc="FlashOn"
  >
    flash-on.svg
  </svg>
  <svg
    data-rc="Flower"
  >
    flower.svg
  </svg>
  <svg
    data-rc="Folder"
  >
    folder.svg
  </svg>
  <svg
    data-rc="Forward"
  >
    forward.svg
  </svg>
  <svg
    data-rc="Forward1"
  >
    forward-1.svg
  </svg>
  <svg
    data-rc="ForwardArrow"
  >
    forward-arrow.svg
  </svg>
  <svg
    data-rc="Gas"
  >
    gas.svg
  </svg>
  <svg
    data-rc="GenderFemale"
  >
    gender-female.svg
  </svg>
  <svg
    data-rc="GenderMale"
  >
    gender-male.svg
  </svg>
  <svg
    data-rc="GenderSymbolFemale"
  >
    gender-symbol-female.svg
  </svg>
  <svg
    data-rc="GenderSymbolMale"
  >
    gender-symbol-male.svg
  </svg>
  <svg
    data-rc="GenderSymbolUni"
  >
    gender-symbol-uni.svg
  </svg>
  <svg
    data-rc="GenderUnisex"
  >
    gender-unisex.svg
  </svg>
  <svg
    data-rc="Gift"
  >
    gift.svg
  </svg>
  <svg
    data-rc="Glasses"
  >
    glasses.svg
  </svg>
  <svg
    data-rc="Globe"
  >
    globe.svg
  </svg>
  <svg
    data-rc="Globe2"
  >
    globe-2.svg
  </svg>
  <svg
    data-rc="GraduationCap"
  >
    graduation-cap.svg
  </svg>
  <svg
    data-rc="Heart"
  >
    heart.svg
  </svg>
  <svg
    data-rc="Hide"
  >
    hide.svg
  </svg>
  <svg
    data-rc="Home"
  >
    home.svg
  </svg>
  <svg
    data-rc="Honeypot"
  >
    honeypot.svg
  </svg>
  <svg
    data-rc="Image"
  >
    image.svg
  </svg>
  <svg
    data-rc="Key"
  >
    key.svg
  </svg>
  <svg
    data-rc="Key2"
  >
    key-2.svg
  </svg>
  <svg
    data-rc="Laptop"
  >
    laptop.svg
  </svg>
  <svg
    data-rc="Layout"
  >
    layout.svg
  </svg>
  <svg
    data-rc="Leaf"
  >
    leaf.svg
  </svg>
  <svg
    data-rc="Link"
  >
    link.svg
  </svg>
  <svg
    data-rc="Loading"
  >
    loading.svg
  </svg>
  <svg
    data-rc="Location"
  >
    location.svg
  </svg>
  <svg
    data-rc="Lock"
  >
    lock.svg
  </svg>
  <svg
    data-rc="LockRound"
  >
    lock-round.svg
  </svg>
  <svg
    data-rc="LockRoundOpen"
  >
    lock-round-open.svg
  </svg>
  <svg
    data-rc="MagicWand"
  >
    magic-wand.svg
  </svg>
  <svg
    data-rc="Mail"
  >
    mail.svg
  </svg>
  <svg
    data-rc="Mallet"
  >
    mallet.svg
  </svg>
  <svg
    data-rc="Map"
  >
    map.svg
  </svg>
  <svg
    data-rc="Megaphone"
  >
    megaphone.svg
  </svg>
  <svg
    data-rc="Menu"
  >
    menu.svg
  </svg>
  <svg
    data-rc="MenuLeft"
  >
    menu-left.svg
  </svg>
  <svg
    data-rc="MenuRight"
  >
    menu-right.svg
  </svg>
  <svg
    data-rc="Microphone"
  >
    microphone.svg
  </svg>
  <svg
    data-rc="Minus"
  >
    minus.svg
  </svg>
  <svg
    data-rc="MoneyNote"
  >
    money-note.svg
  </svg>
  <svg
    data-rc="MoneyNotes"
  >
    money-notes.svg
  </svg>
  <svg
    data-rc="More"
  >
    more.svg
  </svg>
  <svg
    data-rc="MoreVert"
  >
    more-vert.svg
  </svg>
  <svg
    data-rc="Mug"
  >
    mug.svg
  </svg>
  <svg
    data-rc="Multiply"
  >
    multiply.svg
  </svg>
  <svg
    data-rc="MyPension"
  >
    my-pension.svg
  </svg>
  <svg
    data-rc="Newspaper"
  >
    newspaper.svg
  </svg>
  <svg
    data-rc="Notification"
  >
    notification.svg
  </svg>
  <svg
    data-rc="Paste"
  >
    paste.svg
  </svg>
  <svg
    data-rc="Pause"
  >
    pause.svg
  </svg>
  <svg
    data-rc="PenTool"
  >
    pen-tool.svg
  </svg>
  <svg
    data-rc="Percent"
  >
    percent.svg
  </svg>
  <svg
    data-rc="Play"
  >
    play.svg
  </svg>
  <svg
    data-rc="Plug"
  >
    plug.svg
  </svg>
  <svg
    data-rc="Plus"
  >
    plus.svg
  </svg>
  <svg
    data-rc="Power1"
  >
    power-1.svg
  </svg>
  <svg
    data-rc="Printer"
  >
    printer.svg
  </svg>
  <svg
    data-rc="Profile"
  >
    profile.svg
  </svg>
  <svg
    data-rc="Purse"
  >
    purse.svg
  </svg>
  <svg
    data-rc="Puzzle"
  >
    puzzle.svg
  </svg>
  <svg
    data-rc="Refresh"
  >
    refresh.svg
  </svg>
  <svg
    data-rc="Resize"
  >
    resize.svg
  </svg>
  <svg
    data-rc="Rosette"
  >
    rosette.svg
  </svg>
  <svg
    data-rc="Scales"
  >
    scales.svg
  </svg>
  <svg
    data-rc="Scroll"
  >
    scroll.svg
  </svg>
  <svg
    data-rc="Search"
  >
    search.svg
  </svg>
  <svg
    data-rc="Send"
  >
    send.svg
  </svg>
  <svg
    data-rc="Settings"
  >
    settings.svg
  </svg>
  <svg
    data-rc="Shield"
  >
    shield.svg
  </svg>
  <svg
    data-rc="Shift"
  >
    shift.svg
  </svg>
  <svg
    data-rc="ShootingStar"
  >
    shooting-star.svg
  </svg>
  <svg
    data-rc="Shuffle"
  >
    shuffle.svg
  </svg>
  <svg
    data-rc="Snowflake"
  >
    snowflake.svg
  </svg>
  <svg
    data-rc="Snowflake2"
  >
    snowflake-2.svg
  </svg>
  <svg
    data-rc="Sound"
  >
    sound.svg
  </svg>
  <svg
    data-rc="SoundMuted"
  >
    sound-muted.svg
  </svg>
  <svg
    data-rc="SoundPlaying"
  >
    sound-playing.svg
  </svg>
  <svg
    data-rc="Star"
  >
    star.svg
  </svg>
  <svg
    data-rc="StarFill"
  >
    star-fill.svg
  </svg>
  <svg
    data-rc="StarFillHalf"
  >
    star-fill-half.svg
  </svg>
  <svg
    data-rc="Succulent"
  >
    succulent.svg
  </svg>
  <svg
    data-rc="Tag"
  >
    tag.svg
  </svg>
  <svg
    data-rc="ThumbsDown"
  >
    thumbs-down.svg
  </svg>
  <svg
    data-rc="ThumbsUp"
  >
    thumbs-up.svg
  </svg>
  <svg
    data-rc="Time"
  >
    time.svg
  </svg>
  <svg
    data-rc="Triangle"
  >
    triangle.svg
  </svg>
  <svg
    data-rc="Trophy"
  >
    trophy.svg
  </svg>
  <svg
    data-rc="Tv"
  >
    tv.svg
  </svg>
  <svg
    data-rc="UndoAlt"
  >
    undo-alt.svg
  </svg>
  <svg
    data-rc="Upload"
  >
    upload.svg
  </svg>
  <svg
    data-rc="UserAdd"
  >
    user-add.svg
  </svg>
  <svg
    data-rc="Users"
  >
    users.svg
  </svg>
  <svg
    data-rc="View"
  >
    view.svg
  </svg>
  <svg
    data-rc="Wallet"
  >
    wallet.svg
  </svg>
  <svg
    data-rc="Wallet2"
  >
    wallet-2.svg
  </svg>
  <svg
    data-rc="Wifi"
  >
    wifi.svg
  </svg>
  <svg
    data-rc="Wrench"
  >
    wrench.svg
  </svg>
</DocumentFragment>
`;

exports[`Svg SocialIcon should render component correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe lmcdLK"
    data-rc="SocialIcon.Facebook"
  >
    <svg>
      facebook-active.svg
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Svg SocialIcons should render svg correctly 1`] = `
<DocumentFragment>
  <svg
    data-rc="Facebook"
  >
    facebook-active.svg
  </svg>
  <svg
    data-rc="Instagram"
  >
    instagram-active.svg
  </svg>
  <svg
    data-rc="LinkedIn"
  >
    linkedin-active.svg
  </svg>
  <svg
    data-rc="Podcast"
  >
    podcast-active.svg
  </svg>
  <svg
    data-rc="Twitter"
  >
    twitter-active.svg
  </svg>
  <svg
    data-rc="Youtube"
  >
    youtube-active.svg
  </svg>
</DocumentFragment>
`;
